### DISCUSSION TITLE
Error message on HPC but no error on local machine

### DISCUSSION THREAD

**Discussion by xueyang94** [idaholab/moose#21866] [Q&A General] [answered]
I have a remote branch that I have checked out on both local machine and on HPC (of my University). On the local machine, I can compile and run simulations without issue. On HPC, compiling phase_field-opt receives a warning: 

> /usr/bin/ld: warning: libpmi.so.0, needed by /apps/mpi/gcc/9.3.0/openmpi/4.1.1/lib64/libmpi_usempif08.so, may conflict with libpmi.so.1

When running simulations on HPC, I receive an error below. No error on local machine. Any suggestions on how to remove this error? Thanks.

> [c0701a-s3:6130 :0:6130] Caught signal 11 (Segmentation fault: address not mapped to object at address (nil))
==== backtrace (tid:   6130) ====
 0 0x000000000004ee05 ucs_debug_print_backtrace()  ???:0
 1 0x000000000016ab91 __memcmp_sse4_1()  :0
 2 0x0000000000325208 std::_Rb_tree<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::unique_ptr<libMesh::Parameters::Value, std::default_delete<libMesh::Parameters::Value> > >, std::_Select1st<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::unique_ptr<libMesh::Parameters::Value, std::default_delete<libMesh::Parameters::Value> > > >, std::less<void>, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::unique_ptr<libMesh::Parameters::Value, std::default_delete<libMesh::Parameters::Value> > > > >::_M_find_tr<std::basic_string_view<char, std::char_traits<char> >, void>()  ???:0
 3 0x00000000005ac051 libMesh::Parameters::have_parameter<MaterialPropertyName>()  ???:0
 4 0x0000000000a38077 MaterialPropertyInterface::deducePropertyName()  ???:0
 5 0x00000000005f11f4 DerivativeMaterialInterface<JvarMapKernelInterface<Kernel> >::getMaterialPropertyDerivative<double, false>()  ???:0
 6 0x00000000005d5a2f NestKKSSplitCHCRes::NestKKSSplitCHCRes()  ???:0
 7 0x00000000006b2349 Registry::build<NestKKSSplitCHCRes, MooseObject>()  ???:0
 8 0x0000000001b0e4ea Factory::create()  ???:0
 9 0x00000000011c3c5e NonlinearSystemBase::addKernel()  ???:0
10 0x0000000000de1c8e FEProblemBase::addKernel()  ???:0
11 0x000000000140056a AddKernelAction::act()  ???:0
12 0x00000000013f899d Action::timedAct()  ???:0
13 0x0000000001404ed1 ActionWarehouse::executeActionsWithAction()  ???:0
14 0x000000000140534f ActionWarehouse::executeAllActions()  ???:0
15 0x0000000001b1c812 MooseApp::runInputFile()  ???:0
16 0x0000000001b1feef MooseApp::run()  ???:0
17 0x0000000000402a0a main()  ???:0
18 0x0000000000022545 __libc_start_main()  ???:0
19 0x0000000000402c80 _start()  ???:0
=================================
salloc: Relinquishing job allocation 45222753
salloc: Job allocation 45222753 has been revoked.

**Comment by hugary1995** [comment]
The warning seems unrelated to the backtrace.

**Comment by GiudGiud** [comment, marked as answer]
I agree. 
Can you try the following: 
- run in debug mode see if you hit an assert
- run in debug mode with a debugger active and get a full backtrace with line numbers
- replace this ParsedMaterial with a dummy material and see if it runs ok. If so, we should try to modify the JIT parameters of this material

are you running this in parallel?

**Reply by xueyang94** [reply]
I found out that I had a syntax error in a kernel I wrote (NestKKSSplitCHCRes). The problem is solved. Thanks for the debugging info!
